prow/pubsub/subscriber/subscriber.go

```diff
@@ -162,15 +163,21 @@ func (s *Subscriber) handlePeriodicJob(l *logrus.Entry, msg messageInterface, su
 		pj.Status.State = prowapi.ErrorState
 		pj.Status.Description = err.Error()
 		if s.Reporter.ShouldReport(&prowJob) {
-			s.Reporter.Report(&prowJob)
+			if err := s.Reporter.Report(&prowJob); err != nil {
+				l.Warningf("failed to report status. %v", err)
+			}
 		}
 	}
 
 	if err := pe.FromPayload(msg.getPayload()); err != nil {
 		return err
 	}
 	var periodicJob *config.Periodic
-	l.Infof("looking for periodic job %s", pe.Name)
+	jobName := strings.TrimSpace(pe.Name)
+	if jobName == "" {
+		return fmt.Errorf("job name is empty")
+	}
+	l.Infof("looking for periodic job %s", jobName)
```

Comment:
isn't this will fall into `if periodicJob == nil` block below anyway?

Reply:
Good point. Just changed the logging such that it is clear that the string is empty or includes extra white spaces.